### PR TITLE
Add draft text/wgsl media type registration. WGSL uses UTF-8

### PR DESCRIPTION
**Pull request by dneto0**
Fixes: #1682 #565

### CODE REVIEW

**Comment by dneto0** [comment]
For next steps see https://github.com/gpuweb/gpuweb/issues/1682#issuecomment-1286268367

File: wgsl/media-type-registration.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
+Type name: text
+
+Subtype name: wgsl
+
+Required parameters: N/A
+
+Optional parameters: None
+
+Encoding considerations: binary
```
Comment (dneto0) [inline wgsl/media-type-registration.txt:9]: "binary" is used here. The other options are:
7bit: for 7bit text
8bit: for 8bit text separated by CR/LF.  But UTF-8 doesn't fit that.
binary: all 8 bits in each octet must be preserved.  text/javascript uses "binary"
framed:  For framed binary data

File: wgsl/media-type-registration.txt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  Implementations of WebGPU [WebGPU]. This is expected to include
+  web browsers.
+
+Fragment identifier considerations: None
+
+Additional information:
+
+  Magic number(s): None
+  File extension(s): .wgsl
+  Macintosh file type code(s): TEXT
```
Comment (dneto0) [inline wgsl/media-type-registration.txt:46]: For Macintosh file type, I copied what text/javascript has

I guess @litherum can advise

File: wgsl/media-type-registration.txt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+  Implementations of WebGPU [WebGPU]. This is expected to include
+  web browsers.
+
+Fragment identifier considerations: None
+
+Additional information:
+
+  Magic number(s): None
+  File extension(s): .wgsl
```
Comment (dneto0) [inline wgsl/media-type-registration.txt:45]: .wgsl ok with everyone?

**Comment by mehmetoguzderin** [comment]
If I may comment, I think the latest decision was to reflect the consensus about "...ignoring of BOM" (see https://github.com/gpuweb/gpuweb/issues/565#issuecomment-1142583935) with it, but I don't see it in the contents.

Thanks.

File: wgsl/media-type-registration.txt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  Macintosh file type code(s): TEXT
+
+Person & email address to contact for further information:
+  David Neto, dneto@google.com, or  the Editors listed in [WGSL].
```
Comment (annevk) [inline wgsl/media-type-registration.txt:49]:
```suggestion
  David Neto, dneto@google.com, or the Editors listed in [WGSL]. 
```

File: wgsl/media-type-registration.txt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+Additional information:
+
+  Magic number(s): None
+  File extension(s): .wgsl
+  Macintosh file type code(s): TEXT
+
+Person & email address to contact for further information:
+  David Neto, dneto@google.com, or  the Editors listed in [WGSL]. 
+
+Intended usage:  COMMON
```
Comment (annevk) [inline wgsl/media-type-registration.txt:51]:
```suggestion
Intended usage: COMMON
```

File: wgsl/media-type-registration.txt [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+Author:  W3C.  See the Editors listed in [WGSL].
```
Comment (annevk) [inline wgsl/media-type-registration.txt:53]:
```suggestion
Author: W3C. See the Editors listed in [WGSL].
```

File: wgsl/media-type-registration.txt [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+Change controller:  W3C
```
Comment (annevk) [inline wgsl/media-type-registration.txt:55]:
```suggestion
Change controller: W3C
```

**Review by annevk — APPROVED** [review]
Looks good, thanks for fixing this!

**Comment by dneto0** [comment]
> If I may comment, I think the latest decision was to reflect the consensus about "...ignoring of BOM" (see [#565 (comment)](https://github.com/gpuweb/gpuweb/issues/565#issuecomment-1142583935)) with it, but I don't see it in the contents.
> 
> Thanks.

Sorry I missed that.   I had to educate myself about BOM.  I found [this](https://datatracker.ietf.org/doc/html/rfc3629#page-6:~:text=A%20protocol%20SHOULD%20forbid%20use%20of%20U%2BFEFF%20as%20a%20signature%20for%20those%0A%20%20%20%20%20%20textual%20protocol%20elements%20that%20the%20protocol%20mandates%20to%20be%20always%0A%20%20%20%20%20%20UTF%2D8%2C%20the%20signature%20function%20being%20totally%20useless%20in%20those%0A%20%20%20%20%20%20cases.):

> A protocol SHOULD forbid use of U+FEFF as a signature for those
      textual protocol elements that the protocol mandates to be always
      UTF-8, the signature function being totally useless in those
      cases.

WGSL falls into this case.

**Review by mehmetoguzderin — APPROVED** [review]
Thank you very much for your consideration of spelling of BOM with this development!

**Comment by dneto0** [comment]
Discussed in the 2022-10-25 meeting.  There was no opposition.
The only discussion was about BOM, and that has been resolved above.  Thanks @mehmetoguzderin

**Comment by kdashg** [comment]
<details><summary><a href="https://github.com/gpuweb/gpuweb/wiki/wgsl-2022-10-25-minutes#add-draft-textwgsl-media-type-registration-wgsl-uses-utf-8-3542">WGSL 2022-10-25 Minutes</a></summary>

* AB: Happy someone else is happy. Very little opinion, just get it done. Don't know about BOM.
* KG: Sounds like no-one is opposed.

</details>

**Pull request merged**